src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Shape randomShape = Shape.values()[random.nextInt(Shape.values().length)];
+        // This requirement looks strange, but as you wish :-)
+        // ColorSupplier with public String getRandomColor() method - for generating random string,
+        Color randomColor = Color.valueOf(colorSupplier.getRandomColor());
+
+        switch (randomShape) {
+            case CIRCLE:
+                return new Circle(randomColor, random.nextInt(Figure.MAX_SIZE));
+            case ISOSCELES_TRAPEZOID:
+                return new IsoscelesTrapezoid(randomColor,
+                        random.nextInt(Figure.MAX_SIZE),
+                        random.nextInt(Figure.MAX_SIZE),
+                        random.nextInt(Figure.MAX_SIZE));
+            case RECTANGLE:
+                return new Rectangle(randomColor,
+                        random.nextInt(Figure.MAX_SIZE),
+                        random.nextInt(Figure.MAX_SIZE));
+            case RIGHT_TRIANGLE:
+                return new RightTriangle(randomColor,
+                        random.nextInt(Figure.MAX_SIZE),
+                        random.nextInt(Figure.MAX_SIZE));
+            case SQUARE:
+                return new Square(randomColor,
+                        random.nextInt(Figure.MAX_SIZE));
+            default:
+                System.out.println("Something went wrong!");
+                break;
+        }
+        return null;
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE, 10);
```

Comment:
All magic numbers, strings, symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)

Reply:
Added. Thanks!